cibuildwheel/macos.py

```diff
@@ -251,60 +311,140 @@ def build(options: BuildOptions) -> None:
+
+                if config.identifier.endswith('universal2'):
+                    delocate_archs = 'x86_64,arm64'
+                elif config.identifier.endswith('arm64'):
+                    delocate_archs = 'arm64'
+                else:
+                    delocate_archs = 'x86_64'
+
+                repair_command_prepared = prepare_command(
+                    options.repair_command,
+                    wheel=built_wheel,
+                    dest_dir=repaired_wheel_dir,
+                    delocate_archs=delocate_archs,
+                )
                 call(repair_command_prepared, env=env, shell=True)
             else:
                 shutil.move(str(built_wheel), repaired_wheel_dir)
 
             repaired_wheel = next(repaired_wheel_dir.glob('*.whl'))
 
-            if options.test_command:
-                log.step('Testing wheel...')
-                # set up a virtual environment to install and test from, to make sure
-                # there are no dependencies that were pulled in at build time.
-                call(['pip', 'install', 'virtualenv', *dependency_constraint_flags], env=env)
-                venv_dir = Path(tempfile.mkdtemp())
-
-                # Use --no-download to ensure determinism by using seed libraries
-                # built into virtualenv
-                call(['python', '-m', 'virtualenv', '--no-download', venv_dir], env=env)
-
-                virtualenv_env = env.copy()
-                virtualenv_env['PATH'] = os.pathsep.join([
-                    str(venv_dir / 'bin'),
-                    virtualenv_env['PATH'],
-                ])
-
-                # check that we are using the Python from the virtual environment
-                call(['which', 'python'], env=virtualenv_env)
-
-                if options.before_test:
-                    before_test_prepared = prepare_command(options.before_test, project='.', package=options.package_dir)
-                    call(before_test_prepared, env=virtualenv_env, shell=True)
-
-                # install the wheel
-                call(['pip', 'install', str(repaired_wheel) + options.test_extras], env=virtualenv_env)
-
-                # test the wheel
-                if options.test_requires:
-                    call(['pip', 'install'] + options.test_requires, env=virtualenv_env)
-
-                # run the tests from $HOME, with an absolute path in the command
-                # (this ensures that Python runs the tests against the installed wheel
-                # and not the repo code)
-                test_command_prepared = prepare_command(
-                    options.test_command,
-                    project=Path('.').resolve(),
-                    package=options.package_dir.resolve()
-                )
-                call(test_command_prepared, cwd=os.environ['HOME'], env=virtualenv_env, shell=True)
+            if repaired_wheel.stem.endswith('_universal2'):
```

Comment:
Should we have a flag or something to disable this? Or even disable it by default, with a flag to enable? Seems like this is expected to be a very temporary workaround.

Reply:
It's pretty harmless, but even so, I wasn't expecting to merge including this. We're waiting for a pip release that includes https://github.com/pypa/packaging/pull/380 ... so we'd need a packaging/tags release and a pip release it looks like pip is getting close to their 21.0 release, which is slated for some time January 2021. I don't know their policy for updating the vendored dependencies though - do you know if they'll normally bump packaging before an update?

Reply:
This is still an open question, I'm happy to merge it as-is, but hopefully we'll get a pip/packaging update very soon and we can remove the hack entirely.